include/tvm/runtime/device_api.h

```diff
@@ -90,6 +91,16 @@ class TVM_DLL DeviceAPI {
    */
   virtual void* AllocDataSpace(TVMContext ctx, size_t nbytes, size_t alignment,
                                DLDataType type_hint) = 0;
+  /*!
+   * \brief Allocate a data space on device.
+   * \param ctx The device context to perform operation.
+   * \param shape The shape of allocated tensor.
+   * \param dtype The type of elements.
+   * \param mem_scope The memory scope of allocated tensor.
+   * \return The allocated device pointer.
+   */
+  virtual void* AllocDataSpace(TVMContext ctx, std::vector<int64_t> shape, DLDataType dtype,
```

Comment:
let us consider doing ```int ndim, const int64_t* shape,``` instead, to be more compatible with the DLTensor*'s field without constructing a vector.